library/src/main/kotlin/build/buf/connect/ProtocolClientInterface.kt

```diff
@@ -56,6 +56,21 @@ interface ProtocolClientInterface {
         methodSpec: MethodSpec<Input, Output>
     ): ResponseMessage<Output>
 
+    /**
+     * Perform a synchronous unary (non-streaming) request.
+     *
+     * @param request The outbound request message.
+     * @param headers The outbound request headers to include.
+     * @param methodSpec The Method for RPC path.
+     *
+     * @return The ResponseMessage for the unary call.
+     */
+    fun <Input : Any, Output : Any> unarySync(
+        request: Input,
+        headers: Headers,
+        methodSpec: MethodSpec<Input, Output>
+    ): Call<Output>
```

Comment:
It feels a little weird to me to have a `Call` class that is only used when a specific (non-default) option is set to `true`, and isn't used in the happy path. If we want something like this, what about making is specific like `BlockingUnaryCall`?